### PR TITLE
Updated TeamsService to support new ID, also changed some components …

### PR DESCRIPTION
**Pull request by fjgallard**
…that are dependent on the service so that the app does not break.

There are some placeholder code here that only use the first lead memberships they have, but will have to change this with more complex logic to account for switching teams to manage.

There are also probably some bugs caused by this change that flew off the radar, but let's fix it as we find them LOL 💃

### CODE REVIEW

File: src/app/teams/teams.service.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { Injectable } from '@angular/core';
 import { from, Observable } from 'rxjs';
-import { map } from 'rxjs/operators';
+import { map, flatMap, first } from 'rxjs/operators';
```
Comment (nbcp) [inline src/app/teams/teams.service.ts]: unused

File: src/app/teams/teams.service.ts [hunk 2]
```diff
@@ -27,16 +27,54 @@ export class TeamsService {
     }));
   }
 
-  addToTeam(uid: string) {
-    return this.membersCollection.doc<Membership>(uid).update({
+  addToTeam(membership: Membership) {
```
Comment (nbcp) [inline src/app/teams/teams.service.ts]: you don't have to pass the whole Membership object here, only the membership id as string is enough
```suggestion
  addToTeam(id: string) {
```
then you can get rid of 
`const id = membership.uid + membership.teamId;`

File: src/app/teams/teams.service.ts [hunk 3]
```diff
@@ -65,7 +103,7 @@ export class TeamsService {
   }
 
   getTeamId(uid: string): Observable<string> {
```
Comment (nbcp) [inline src/app/teams/teams.service.ts]: unused, please remove

File: src/app/teams/teams.service.ts [hunk 4]
```diff
@@ -76,13 +114,15 @@ export class TeamsService {
     );
   }
 
+  // need to change this
   setAsLead(uid: string, isLead: boolean) {
```
Comment (nbcp) [inline src/app/teams/teams.service.ts]: unused method, please remove

File: src/app/teams/teams.service.ts [hunk 5]
```diff
@@ -76,13 +114,15 @@ export class TeamsService {
...
     this.membersCollection.doc<Membership>(uid).update({
       isLead: isLead
     });
   }
 
-  removeTeamMember(uid: string) {
-    return this.membersCollection.doc<Membership>(uid).delete();
+  removeTeamMember(membership: Membership) {
```
Comment (nbcp) [inline src/app/teams/teams.service.ts]:
```suggestion
  removeTeamMember(id: string) {
```

File: src/app/teams/teams.service.ts [hunk 6]
```diff
@@ -27,16 +27,54 @@ export class TeamsService {
...
+  addToTeam(membership: Membership) {
+    const id = membership.uid + membership.teamId;
+    return this.membersCollection.doc<Membership>(id).update({
       isApproved: true
     });
   }
 
-  getMembership(uid: string): Observable<Membership> {
+  // Deprecated
+  getMembership_(uid: string): Observable<Membership> {
```
Comment (nbcp) [inline src/app/teams/teams.service.ts]: I think we are over-complicating this change, this ticket is actually a preparation for #9.
To minimize changes and maintain backwards compatibility, the simplest approach would be to make getMembership() smarter. Make it return only one team with priority to those of which the user is team lead of.

Aside from that method, I think we only need getAllMemberships() to be able to get the list of all teams that we'll be using for #9.

Comment (fjgallard) [reply src/app/teams/teams.service.ts]: The "new" getMembership() function is actually just renamed as "getPlayerMembership()"(which has the behavior described) just to be able to more clearly distinguish between getting a membership as a player, and getting membership as a lead. Would it be preferred to keep the old naming?

Comment (fjgallard) [reply src/app/teams/teams.service.ts]: Moreover, in preparation for #9 as well, there is a "getLeadMemberships()" function in the service as well, which, as the name implies, gets all Memberships the player is a Lead of. A separate "getAllMemberships()" function that gets literally all the memberships, player or lead, was made, though without purpose as of now.

**Pull request closed without merging**